scipy/stats/_sensitivity_analysis.py

```diff
@@ -0,0 +1,588 @@
+        + 0.1 * (x[:, 2]**4) * np.sin(x[:, 0])
+    )
+    return f_eval.reshape(-1, 1)
+
+
+def sample_A_B(
+    n: IntNumber,
+    dists: List[PINVDist | PPFDist],
+    random_state: SeedType = None
+) -> Tuple[np.ndarray, np.ndarray]:
+    """Sample two matrices A and B.
+
+    Uses a Sobol' sequence with 2`d` columns to have 2 uncorrelated matrices.
+    This is more efficient than using 2 random draw of Sobol'.
+    See sec. 5 from [1]_.
+
+    References
+    ----------
+    .. [1] Saltelli, A., P. Annoni, I. Azzini, F. Campolongo, M. Ratto, and
+       S. Tarantola. "Variance based sensitivity analysis of model
+       output. Design and estimator for the total sensitivity index."
+       Computer Physics Communications, 181(2):259-270,
+       :doi:`10.1016/j.cpc.2009.09.018`, 2010.
+    """
+    d = len(dists)
+    A_B = qmc.Sobol(d=2*d, seed=random_state, bits=64).random(n)
+
+    A, B = A_B[:, :d], A_B[:, d:]
+
+    for d_, dist in enumerate(dists):
+        if not hasattr(dist, 'ppf'):
+            dist = NumericalInversePolynomial(
+                dist, random_state=random_state
+            )
+        A[:, d_] = dist.ppf(A[:, d_])
+        B[:, d_] = dist.ppf(B[:, d_])
+
+    return A, B
+
+
+def sample_AB(A: np.ndarray, B: np.ndarray) -> np.ndarray:
+    """AB matrix.
+
+    AB: columns of B into A. Shape (n*d, d). e.g in 2d
+    Take A and replace 1st column with 1st column of B. You have (n, d)
+    Then A and replace 2nd column with 2nd column of B. You have (n, d)
+    Concatenate to get AB. Which means AB is (2*n, d).
+    """
+    n, d = A.shape
+    AB = np.tile(A, (d, 1, 1))
+    i = np.arange(d)
+    AB[i, :, i] = B[:, i].T
+    AB = AB.reshape(-1, d)
+    return AB
+
+
+def saltelli_2010(
+    f_A: np.ndarray, f_B: np.ndarray, f_AB: np.ndarray
+) -> Tuple[np.ndarray, np.ndarray]:
+    r"""Saltelli2010 formulation.
+
+    .. math::
+
+        S_i = \frac{1}{N} \sum_{j=1}^N
+        f(\mathbf{B})_j (f(\mathbf{AB}^{(i)})_j - f(\mathbf{A})_j)
+
+    .. math::
+
+        S_{T_i} = \frac{1}{N} \sum_{j=1}^N
+        (f(\mathbf{A})_j - f(\mathbf{AB}^{(i)})_j)^2
+
+    Parameters
+    ----------
+    f_A, f_B, f_AB : array_like (n, x)
+        Function evaluations ``n`` being the number of evaluations and ``x``
+        a vector output.
+
+    Returns
+    -------
+    s, st : array_like (x, d)
+        First order and total order Sobol' indices.
+
+    References
+    ----------
+    .. [1] Saltelli, A., P. Annoni, I. Azzini, F. Campolongo, M. Ratto, and
+       S. Tarantola. "Variance based sensitivity analysis of model
+       output. Design and estimator for the total sensitivity index."
+       Computer Physics Communications, 181(2):259-270,
+       :doi:`10.1016/j.cpc.2009.09.018`, 2010.
+    """
+    f_AB = f_AB.reshape(-1, *f_A.shape)
+
+    var = np.var(np.vstack([f_A, f_B]), axis=0)
+
+    s = np.mean(f_B * (f_AB - f_A), axis=1) / var
+    st = 0.5 * np.mean((f_A - f_AB) ** 2, axis=1) / var
```

Comment:
```suggestion
    st = 0.5 * np.mean((f_A - f_AB) ** 2, axis=1) / var  # Table 2 (f)
```

Reply:
Please also add a note about where `var` above is defined in the paper and where it says to divide these quantities by it.